aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -156,6 +157,42 @@ storage:
             --net=host \
             --dns=host \
             --exec=/bootkube -- start --asset-dir=/assets "$@"
+    - path: /opt/etcd-rejoin
+      filesystem: root
+      mode: 0555
+      contents:
+        inline: |
+          #!/bin/bash
+          set -eou pipefail
+          # Script to rejoin as fresh node to an existing etcd cluster when the node cannot join.
+          # It works around the etcd limitation (by design) described in https://github.com/etcd-io/etcd/issues/5422
+          # The script should be added to /etc/systemd/system/etcd-member.service.d/40-etcd-cluster.conf as ExecStopPost=-/opt/etcd-rejoin
+          # because it makes use of the environment variables set there for etcd-member.service.
+          # It only runs when etcd failed to start, indicating that it cannot join the cluster (e.g., after repovisioning the node or after
+          # the node was down for too long and got kicked out).
+          if [ "$SERVICE_RESULT" != "exit-code" ] || [ "$EXIT_CODE" != "exited" ] || [ "$EXIT_STATUS" = "0" ]; then exit 0; fi
+          # Do not run when etcd failed to start because the node was not fully provisioned yet.
+          if [ ! -d "/etc/ssl/etcd/" ];	then exit 0; fi
+          export ETCDCTL_API=3
+          urls=$(echo "$ETCD_INITIAL_CLUSTER" | tr "," "\n" | cut -d "=" -f 2 | tr "\n" "," | head -c -1)
+          # Double $$ for terraform
+          endpoints="$${urls//2380/2379}"
+          # Assemble etcdctl arguments
+          ARGS="--cacert=/etc/ssl/etcd/etcd-client-ca.crt --cert=/etc/ssl/etcd/etcd-client.crt --key=/etc/ssl/etcd/etcd-client.key --endpoints=$endpoints"
+          # Find old member ID and remove that member
+          ID=$(etcdctl member list $ARGS | grep "$ETCD_NAME" | cut -d , -f 1)
+          etcdctl member remove "$ID" $ARGS
```

Comment:
When a node was down for too long wouldn't we still have the same problem because it kept it's name? At some point the old name also has to be removed and the new member added to the list.
You hinted me to skip removal if it got removed already (manually or for whatever reason).

Reply:
Yeah make sense we can keep it!